spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -217,5 +231,78 @@ private TaskDefinition updateTaskProperties(TaskDefinition taskDefinition) {
 		return builder.build();
 	}
 
+	@Override
+	public void saveTaskDefinition(String name, String dsl) {
+		TaskParser taskParser = new TaskParser(name,
+				dsl, true, true);
+		TaskNode taskNode = taskParser.parse();
+		if(taskNode.isComposed()) {
+			//Create the child task definitions needed for the composed task
+			taskNode.getTaskApps().stream().forEach(task -> {
+				//Add arguments to child task definitions
+				String generatedTaskDSL = task.getName() +
+						task.getArguments().entrySet().stream()
+						.map(argument->String.format(" --%s=%s",
+								argument.getKey() ,argument.getValue()))
+				.collect(Collectors.joining());
+				TaskDefinition composedTaskDefinition = new TaskDefinition(
+						task.getExecutableDSLName(), generatedTaskDSL);
+				saveStandardTaskDefinition(composedTaskDefinition);
+			});
+			taskDefinitionRepository.save(
+					new TaskDefinition(
+							name,
+							createComposedTaskDefinition(
+									taskNode.toExecutableDSL())));
+		}
+		else {
+			saveStandardTaskDefinition(new TaskDefinition(name, dsl));
+		}
+
+	}
+
+	private void saveStandardTaskDefinition(TaskDefinition taskDefinition) {
+		String appName = taskDefinition.getRegisteredAppName();
+		if (registry.find(appName, ApplicationType.task) == null) {
+			throw new IllegalArgumentException(String.format(
+					"Application name '%s' with type '%s' does not exist in the app registry.",
+					appName, ApplicationType.task));
+		}
+		taskDefinitionRepository.save(taskDefinition);
+	}
+
+	private String createComposedTaskDefinition(String graph) {
+		return String.format("%s --graph=\"%s\"",
+				composedTaskProperties.getTaskName(), graph);
+	}
+
+	@Override
+	public void deleteTaskDefinition(String name) {
+		TaskDefinition taskDefinition = taskDefinitionRepository.findOne(name);
+		if (taskDefinition == null) {
+			throw new NoSuchTaskDefinitionException(name);
+		}
+		//if composed-task-runner definition then destroy all child tasks associated with it.
+		if(taskDefinition.getDslText().startsWith(composedTaskProperties.getTaskName()))
+		{
+			String childTaskPrefix = String.format("_%s_",name);
+			taskDefinitionRepository.findAll().forEach(
+					childDefinition -> {
+						if(childDefinition.getName().startsWith(childTaskPrefix)) {
+							destroyTask(childDefinition.getName());
+						}
+					}
+			);
+		}
+		//destroy normal task or composed-task
+		destroyTask(name);
+	}
+
+	private void destroyTask(String name) {
+		TaskDefinition taskDefinition = taskDefinitionRepository.findOne(name);
```

Comment:
It would be safe to throw `NoSuchTaskDefinitionException` if `taskDefinition` is null here as the `destroyTask` is called for child definitions as well.